packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -2902,6 +2926,11 @@ function beginWork(
             // whether to retry the primary children, or to skip over it and
             // go straight to the fallback. Check the priority of the primary
             // child fragment.
+
+            // TODO: Around here, we need to detect that it's "2nd phase" of an
+            // avoided render (maybe just effectTag & SoftCapture ?),
+            // and just skip work for children altogether.
```

Comment:
I think this comment is referring to something like resuming? But I'm not sure we have a reasonable safe way to do that atm or if it's even desirable. Another approach is to commit the children early and in that case there is no work to perform so it's automatic. Maybe just drop the comment until we have a proper plan.